pkg/server/config.go

```diff
@@ -56,7 +64,16 @@ func (c Config) Complete() (*server, error) {
 		return nil, fmt.Errorf("unable to construct a client to connect to the kubelets: %v", err)
 	}
 	nodes := informer.Core().V1().Nodes()
-	scrape := scraper.NewScraper(nodes.Lister(), kubeletClient, c.ScrapeTimeout)
+	ns := strings.TrimSpace(c.NodeSelector)
+	if ns != "" {
+		if nodeSelector, err := labels.Parse(ns); err == nil {
```

Comment:
Please handle errors correctly. We should not just ignore user provided selectors if they make a typo.

Reply:
done